corehq/apps/hqadmin/service_checks.py

```diff
@@ -0,0 +1,170 @@
+"""
+A collection of functions which test the most basic operations of various services.
+"""
+from collections import namedtuple
+import datetime
+import json
+import logging
+from StringIO import StringIO
+import time
+
+from django.core import cache
+from django.conf import settings
+from django.contrib.auth.models import User
+from restkit import Resource
+import requests
+from soil import heartbeat
+
+from corehq.apps.app_manager.models import Application
+from corehq.apps.change_feed.connection import get_kafka_client_or_none
+from corehq.apps.es import GroupES
+from corehq.blobs import get_blob_db
+from corehq.elastic import send_to_elasticsearch
+from corehq.util.decorators import change_log_level
+
+ServiceStatus = namedtuple("ServiceStatus", "success msg")
+
+
+def check_redis():
+    if 'redis' in settings.CACHES:
+        import redis
+        rc = cache.caches['redis']
+        redis_api = redis.StrictRedis.from_url('%s' % rc._server)
+        memory = redis_api.info()['used_memory_human']
+        result = rc.set('serverup_check_key', 'test')
+        return ServiceStatus(result, "Redis is up and using {} memory".format(memory))
+    else:
+        return ServiceStatus(False, "Redis is not configured on this system!")
+
+
+def check_rabbitmq():
+    if settings.BROKER_URL.startswith('amqp'):
+        amqp_parts = settings.BROKER_URL.replace('amqp://', '').split('/')
+        mq_management_url = amqp_parts[0].replace('5672', '15672')
+        vhost = amqp_parts[1]
+        try:
+            mq = Resource('http://%s' % mq_management_url, timeout=2)
+            vhost_dict = json.loads(mq.get('api/vhosts', timeout=2).body_string())
+            for d in vhost_dict:
+                if d['name'] == vhost:
+                    return ServiceStatus(True, 'RabbitMQ OK')
+            return ServiceStatus(False, 'RabbitMQ Offline')
+        except Exception as e:
+            return ServiceStatus(False, "RabbitMQ Error: %s" % e)
+    else:
+        return ServiceStatus(False, "RabbitMQ Not configured")
+
+
+def check_pillowtop():
+    return ServiceStatus(False, "Not implemented")
+
+
+@change_log_level('kafka.client', logging.WARNING)
+def check_kafka():
+    client = get_kafka_client_or_none()
+    if not client:
+        return ServiceStatus(False, "Could not connect to Kafka")
+    return ServiceStatus(True, "Kafka's fine. Probably.")
```

Comment:
I probably should have put a TODO here...  This one and pillowtop still need work